src/test/java/io/managed/services/test/rhoas/KafkaRhoasAclTests.java

```diff
@@ -0,0 +1,245 @@
+        assertTrue(acl1.isPresent(), "ACL created must exist");
+        assertTrue(acl2.isPresent(), "ACL created must exist");
+        assertTrue(acl3.isPresent(), "ACL created must exist");
+        assertTrue(acl4.isPresent(), "ACL created must exist");
+        assertTrue(acl5.isPresent(), "ACL created must exist");
+        assertTrue(acl6.isPresent(), "ACL created must exist");
+    }
+
+    // Primary user deletes six ACLs to remove grant-access permission to produce and consume records on a topic
+    // Outcome: Six ACLs assigned to entity are deleted and can NOT be found once it is searched for based on entity
+    @SneakyThrows
+    @Test(dataProvider = "aclEntityPairs", dependsOnMethods = "testCreateAclGrantAccess", enabled = true)
+    public void testDeleteAclGrantAccess(ACLEntityType entityType, String entityIdentificator) {
+        LOGGER.info("Delete grant-access ACL for {}", entityType.name);
+        AclBindingListPage aclList;
+
+        cli.deleteAcl(entityType, entityIdentificator, AclOperation.READ, AclPermissionType.ALLOW);
+        aclList = cli.listACLs(entityType, entityIdentificator);
+        Optional<AclBinding> acl1 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.READ, AclPermissionType.ALLOW, AclResourceType.GROUP, ALL_RESOURCES);
+        Optional<AclBinding> acl2 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.READ, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
+        assertTrue(acl1.isEmpty(), "ACL deleted must be removed");
+        assertTrue(acl2.isEmpty(), "ACL deleted must be removed");
+
+        cli.deleteAcl(entityType, entityIdentificator, AclOperation.DESCRIBE, AclPermissionType.ALLOW);
+        aclList = cli.listACLs(entityType, entityIdentificator);
+        Optional<AclBinding> acl3 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.DESCRIBE, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
+        Optional<AclBinding> acl4 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.DESCRIBE, AclPermissionType.ALLOW, AclResourceType.TRANSACTIONAL_ID, ALL_RESOURCES);
+        assertTrue(acl3.isEmpty(), "ACL deleted must be removed");
+        assertTrue(acl4.isEmpty(), "ACL deleted must be removed");
+
+        cli.deleteAllAcls(entityType, entityIdentificator);
+        aclList = cli.listACLs(entityType, entityIdentificator);
+        Optional<AclBinding> acl5 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.CREATE, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
+        Optional<AclBinding> acl6 = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.WRITE, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
+        assertTrue(acl5.isEmpty(), "ACL deleted must be removed");
+        assertTrue(acl6.isEmpty(), "ACL deleted must be removed");
+    }
```

Comment:
I agree that we are testing deletion of ACL more than, but I thought it's better to do it this way because tests definition are more consistent in all the cases: create, list, search and delete. Do you prefer to test deletion only once and remove ACLs for the rest of the creation tests?

I will improve the flow of delete, list, search and assert ACLs.

Reply:
I would be in favour of it if there was some command like `grand-access --revoke`  but here we are instead only deleting one ACL for admin and test 6 deletion in case of producer/consumer.